bibliopixel/led/cube.py

```diff
@@ -0,0 +1,93 @@
+import math, threading, time
+
+from .. import colors, data_maker, font, matrix
+from . base import LEDBase
+from . multimap import MatrixRotation, mapGen, MultiMapBuilder
+
+
+class LEDCube(LEDBase):
+
+    def __init__(self, drivers, x, y, z, coordMap=None, threadedUpdate=False, masterBrightness=255, maker=data_maker.MAKER):
+        super().__init__(drivers, threadedUpdate, masterBrightness, maker)
+
+        self.x = x
+        self.y = y
+        self.z = z
+
+        # self.pixelSize = pixelSize
+        # pw, ph = self.pixelSize
+
+        if self.x * self.y * self.z != self.numLEDs:
+            raise TypeError("(x * y * z) MUST equal the total pixel count!")
+
+        if coordMap:
+            self.matrix_map = coordMap
+        # else:
+        #     if len(self.drivers) == 1:
+        #         self.matrix_map = mapGen(self.width, self.height, serpentine)
+        #     else:
+        #         raise TypeError(
+        #             "Must provide coordMap if using multiple drivers!")
+
+        self.texture = None
+        self.set = self._setColor
+        self._set = self.__setNormal
+
+        # if pw < 0 or pw > self.width or ph < 0 or ph > self.height:
+        #     raise ValueError(
+        #         "pixelSize must be greater than 0 and not larger than total matrix!")
+        # if self.width % pw != 0 or self.height % ph != 0:
+        #     raise ValueError(
+        #         "pixelSize must evenly divide into matrix dimensions!")
+
+        # if pw == 1 and ph == 1:
+        #     self._set = self.__setNormal
+        # else:
+        #     self._set = self.__setScaled
+        #     self.width = self.width / pw
+        #     self.height = self.height / ph
+        #     self.numLEDs = self.width * self.height
+
+    def __setNormal(self, x, y, z, color):
```

Comment:
This is copied over from LEDMatrix. This is not new. If you want to remove that, open an issue ;) But there's already precedent for this being here from the previous code and I'd rather review truly *new* code.